### ISSUE TITLE
Berserking Enchant PPM for Ret Paladin

### ISSUE THREAD

**Issue by Bordiran** [SunwellTracker/issues#1072]
**Decription:** 
The Berserking Enchant should have a high uptime on Ret Paladin because of the way Seal of Vengeance's "hidden strikes" mechanic interacts with the enchant. It seems that the Berserking Enchant has a much lower proc chance than intended on the server because I tested on a dummy for around 10 minutes, actively doing my rotation and had a very low uptime on the Berserking Enchant. 

**How it works:** 
The Berserking Enchant seems to have a very low PPM on this server, especially noticeable when Ret Paladins should have really high uptime on the enchant due to the way Berserking enchant interacts with the "hidden strikes" mechanic of Seal of Vengeance.

**How it should work:** 
Berserking should have a higher uptime and PPM, especially as Ret Paladin due to the way our Seal of Vengeance "hidden strikes" mechanic increases the PPM of the enchant.

I talked to an Officer in my Guild who is playing dual-wield Enhancement Shaman and he has Berserking on both of his 1H weapons and he was telling me that the proc chance of Berserking was very low when he was testing on dummy and the PPM was much lower than it should be while in raid and Enhancement Shaman with double Berserking should have very high uptime with Berserking because they attack so quickly. That led to me testing it out on my Ret Paladin which should have a high uptime on it as well and that's how I discovered the PPM of Berserking is much much lower than intended on the server. It appears to be a problem with the enchant itself and it's PPM because I noticed Berserking would proc from the "hidden strikes" mechanic but the proc chance of the enchant much lower than it should be.

**Source (you should point out proofs of your report, please give us some source):** 

**PROOF:** http://www.worldoflogs.com/reports/rt-ckge2l8lqplzf282/details/6/ - My testing on a dummy with the Berserking Enchant on my Ret Paladin (go to Buffs Cast tab and look at Berserk). This is on a dummy without 20% melee Haste or Bloodlust so it shouldn't be at 75%~ uptime, but it should be around 50%~ uptime at least without those buffs, especially after 10 minutes of testing.

http://www.worldoflogs.com/reports/pt6hgjymsmjqt24s/details/3/?s=2717&e=2899 - Retail Logs from Method ICC HC Ret Paladin on Festergut HC showing Berserk Enchant uptime at 76.6% (go to Buffs Cast tab and look at Berserk).

http://www.worldoflogs.com/reports/rt-g96mnptwpdekznce/details/41/?s=680&e=934 - Enhancement Shaman with double Berserking Enchant on both his 1H weapons having 36.8% uptime with Berserking enchant on Grobbulus (go to Buffs Cast tab and look at Berserk). 

http://web.archive.org/web/20100507223304/http://elitistjerks.com/f76/t68951-retribution_updated_3_3_a/#Enchants - Elitistjerks thread that contains explanation on why Berserking enchant has high uptime for Ret Paladin.

Quote from the thread about the Berserking Enchant

> **How berserking works: Berserking is a ppm based proc, similar to mongoose - in a raid environment it has a ~75% uptime. It can only trigger off attacks flagged as melee attacks and melee specials (Judgements, CS, DS, white swings). It has increased uptime since 3.2 due to SoV's hidden strike mechanic, which seems to be able to trigger and refresh the buff. Because of this, this is our only real choice for any dps weapon enchant.**

http://web.archive.org/web/20100507223304/http://elitistjerks.com/f76/t68951-retribution_updated_3_3_a/#Seals - Explaining how "hidden strikes" works from Seal of Vengeance and how it affects Berserking enchants PPM. 

Quote from the thread regarding Hidden Strikes and how it interacts with the berserking enchant

> When an auto-attack lands (does not dodge/parry/miss) that can proc a seal the of the following things happen independently of each other (see 2 roll system). 1) A "hidden strike" which uses melee combat mechanics occurs. If it lands it refreshes/stacks SoV DoT. Only white swings can trigger a refresh or stack. **(This hidden strike mechanic can also proc things like berserking..)** 2) A weapon damage based proc will occur if you used a special (CS/DS/judge) or if you have a 5 stack (from auto attacks). This attack can not be avoided. Remember #2 happens regardless of #1 landing, it just requires the initial attack (autos, cs, etc) to land. 

Source: http://web.archive.org/web/20100507223304/http://elitistjerks.com/f76/t68951-retribution_updated_3_3_a/#Seals#ixzz5Atck2prW

**Comment by Bordiran** [comment]
As you can see from my testing on the dummy I had 20% uptime on Berserking (221.8s of Uptime out of 1100s of active combat uptime). That's WAY lower than it should be. Even outside of a raid environment. So the proc chance of the Berserking enchant is most definitely bugged on the server which is huge bug and damage loss for every class/spec using Berserking enchant.

**Comment by Bordiran** [comment]
http://www.worldoflogs.com/reports/rt-vv0plupiytqit1nd/details/22/?s=4337&e=4493 and http://www.worldoflogs.com/reports/rt-vv0plupiytqit1nd/details/22/?s=2524&e=2644 from a Raid I did with Berserking, had 20% Melee haste and bloodlust and below 50% uptime for both. Very very low uptime compared to how it should be.

**Comment by knasen88** [comment]
it has a 1 PPM, which seems to work, crunching the numbers with your jawbone, you have 1/16.7 % proc per swing and you did 89 melee abilities and it procced 5 times which fits the math perfectly. The thing that's not working not the enchant but it seems to be this "hidden swing" you mention that is not working. Now to find correct data for how that really worked won't be easy. But i guess this is why abom in a jar was also bis for rets

**Comment by Bordiran** [comment]
The "hidden strikes" mechanic of Seal of Vengeance is proccing the Berserking enchant because I tested on dummy, however it's not proccing it anywhere close to how much it should be. You're right about the Tiny Abom thing though, the hidden strikes generated from proccing Seal of Vengeance is what makes the trinket byfar the best for Ret over other classes/specs and it's also supposed to be the thing that makes Berserking an insane enchant with high uptime for Ret Pala.

So from that testing I concluded that it has to be the enchant itself that is bugged, because if you know how Blizzard calculates PPM it's based purely off Auto Attacks. So if I was only Auto Attacking and I got 1 PPM then it would be correct, but that fact that I'm Auto Attacking, using my abilities, and have the hidden strikes portion as Ret Paladin is absurdly low. It's like when people try to calculate the Drain Life from Bryntroll and see that it has 2 PPM but when you use it you'll see that it's higher than 2 PPM, much higher especially if you're a Ret Paladin with hidden strikes from SoV and Tiny Abom because both proc the Drain Life.

It's obvious that Berserking uptime is bugged when it's uptime should be 75%~ on Ret Paladin and I'm constantly getting 40%~ on my Ret in a raid environment with Melee Haste, Bloodlust, 354 (14%) Haste. The hidden strikes from Seal of Vengeance/Corruption are proccing it, but it is proccing it very rarely when it should be proccing it constantly. That might be what the issue is because I notice sometimes as well that my Trinkets (DMC Greatness and Bandit's Insignia), which should be proccing instantly when their Internal Cooldown are up, are taking quite a bit of time to proc when they're not on ICD.

Idk, could be a number of issues. Definitely need a Developer to look into this issue and see what it is.

**Comment by knasen88** [comment]
No i did the math earlier, without accounting for the hidden strikes mechanic the berserk proc is working as intended. If anything is broke it is the hidden strikes not proccing stuff.

**Comment by Bordiran** [comment]
Yeah I'm still having 50-60% uptime on Berserking in raids on my Ret, pretty bad :/

**Comment by MVMnn** [comment]
This issue is correlated with [https://github.com/SunwellTracker/issues/issues/1663](url), but the initial problem is that hidden strikes do not work correctly.
